apps/community/src/components/sign-in-form.tsx

```diff
@@ -0,0 +1,58 @@
+'use client';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm, useWatch } from 'react-hook-form';
+import type { z } from 'zod';
+
+import { AuthButton } from '~/components/auth-button';
+import { AuthInput } from '~/components/auth-input';
+import * as styles from '~/components/sign-in-form.css';
+import { defaultValues, signInFormSchema } from '~/schemas/sign-in-form-schema';
+
+function SignInForm() {
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<z.infer<typeof signInFormSchema>>({
+    resolver: zodResolver(signInFormSchema),
+    defaultValues,
+  });
+
+  const onSubmit = (data: z.infer<typeof signInFormSchema>) => {
+    console.log(data);
+  };
+
+  const [studentId, password] = useWatch({
+    control,
+    name: ['studentId', 'password'],
+  });
+  const isFormValid = studentId && password;
```

Comment:
```suggestion
  const {
    register,
    handleSubmit,
    control,
    formState: { errors },
  } = useForm<z.infer<typeof signInFormSchema>>({
    resolver: zodResolver(signInFormSchema),
    defaultValues,
  });
  const [studentId, password] = useWatch({
    control,
    name: ['studentId', 'password'],
  });

  const onSubmit = (data: z.infer<typeof signInFormSchema>) => {
    console.log(data);
  };
  const isFormValid = studentId && password;
```

훅은 상단에 선언되어 모여있도록 하면 훅과 이벤트를 구분하고 가독성 있게 구조적으로 볼 수 있어요